### ISSUE TITLE
Fail to test/sslapitest.c when use option DOPENSSL_TLS_SECURITY_LEVEL=0 in OpenSSL 3.4.0

### ISSUE THREAD

**Issue by phongnguyen1023** [openssl/openssl#26255]
When build OpenSSL 3.4.0 with option DOPENSSL_TLS_SECURITY_LEVEL=0. In step make test, an error occurred with test/sslapitest.c

```
# INFO:  @ test/helpers/ssltestlib.c:1289
# SSL_accept() failed -1, 11
# ERROR: (memory) 'p == expected_ciphers' failed @ test/sslapitest.c:735
# --- p
# +++ expected_ciphers
# 0000:-009dc02c00ff
# 0000:+009dc02c
# 
# INFO:  @ test/helpers/ssltestlib.c:1289
# SSL_accept() failed -1, 1
# 40A71ED6677F0000:error:0A0000EA:SSL routines:tls_early_post_process_client_hello:callback failed:ssl/statem/statem_srvr.c:1710:
# INFO:  @ test/helpers/ssltestlib.c:1255
# SSL_connect() failed -1, 1
# 40A71ED6677F0000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:ssl/record/rec_layer_s3.c:908:SSL alert number 80
# ERROR: (bool) 'create_ssl_connection(serverssl, clientssl, SSL_ERROR_NONE) == true' failed @ test/sslapitest.c:779
# false
# OPENSSL_TEST_RAND_SEED=1734598103
not ok 26 - test_client_hello_cb
```

I don't know what this error means, please help me claim it and how to fix.
I noticed there is a difference with the old OpenSSL version in
https://github.com/openssl/openssl/commit/972ee925b16fc3bc7ec71080c439e669754235ab#diff-ac041dcce48346a57d70403803133a58fb4ac5764e173d7cbfd6db2c2df5afdc

**Comment by mattcaswell** [comment]
This is a bug in our tests. Fix in #26256.

**Comment by phongnguyen1023** [comment]
I tested it, and the bug is fixed.
